docs/Coding_Style.md

```diff
@@ -0,0 +1,148 @@
+# FoundationDB Record Layer Coding Style
+
+This document is very much a work in progress, but the intention is to detail rules
+both our code formatting style rules, as well as general paradigms that we follow
```

Comment:
> both our code formatting style rules

Is this true? It seems like our code formatting should (as much as possible) be in the automatic code styling rules, though maybe that's not always possible. It seems like this document is more about practices for general paradigms which are harder to enforce automatically (though maybe not impossible).

Reply:
You are correct, I switched it to a "Best Practices" slant.